stubs/PyAutoGUI/pyautogui/__init__.pyi

```diff
@@ -0,0 +1,233 @@
+import collections.abc
+from _typeshed import Incomplete
+from collections.abc import Callable, Generator
+from datetime import datetime
+from typing import NamedTuple, TypeVar, overload
+from typing_extensions import ParamSpec
+
+# from pyscreeze import Box
+
+class PyAutoGUIException(Exception): ...
+class FailSafeException(PyAutoGUIException): ...
+class ImageNotFoundException(PyAutoGUIException): ...
+
+collectionsSequence = collections.abc.Sequence
+
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
+# TODO: Complete types with pyscreeze once we can import non-types dependencies
+# See: https://github.com/python/typeshed/issues/5768
+
+class _Box(NamedTuple):
+    left: int
+    top: int
+    width: int
+    height: int
+
+def raisePyAutoGUIImageNotFoundException(wrappedFunction: Callable[_P, _R]) -> Callable[_P, _R]: ...
+def locate(*args: Incomplete, **kwargs: Incomplete) -> _Box | None: ...
+def locateAll(*args: Incomplete, **kwargs: Incomplete) -> Generator[_Box, None, None]: ...
+def locateAllOnScreen(*args: Incomplete, **kwargs: Incomplete) -> Generator[_Box, None, None]: ...
+def locateCenterOnScreen(*args: Incomplete, **kwargs: Incomplete) -> Point | None: ...
+def locateOnScreen(*args: Incomplete, **kwargs: Incomplete) -> _Box | None: ...
+def locateOnWindow(*args: Incomplete, **kwargs: Incomplete) -> _Box | None: ...
+def mouseInfo() -> None: ...
+def useImageNotFoundException(value: bool | None = ...) -> None: ...
+
+KEY_NAMES: list[str]
+KEYBOARD_KEYS: list[str]
+LEFT: str
+MIDDLE: str
+RIGHT: str
+PRIMARY: str
+SECONDARY: str
+QWERTY: str
+QWERTZ: str
+
+def isShiftCharacter(character: str) -> bool: ...
+
+MINIMUM_DURATION: float
+MINIMUM_SLEEP: float
+PAUSE: float
+DARWIN_CATCH_UP_TIME: float
+FAILSAFE: bool
+FAILSAFE_POINTS: list[tuple[int, int]]
+LOG_SCREENSHOTS: bool
+LOG_SCREENSHOTS_LIMIT: int
+G_LOG_SCREENSHOTS_FILENAMES: list[str]
+
+class Point(NamedTuple):
+    x: float
+    y: float
+
+class Size(NamedTuple):
+    width: int
+    height: int
+
+def getPointOnLine(x1: float, y1: float, x2: float, y2: float, n: float) -> tuple[float, float]: ...
+def linear(n: float) -> float: ...
+def position(x: int | None = ..., y: int | None = ...) -> Point: ...
+def size() -> Size: ...
+@overload
+def onScreen(xy: tuple[float, float]) -> bool: ...
```

Comment:
The function does still, technically, work if you explicitly pass `None` in as the second argument:

```suggestion
def onScreen(x: tuple[float, float], y: None = ...) -> bool: ...
```